src/lib/mina_ledger/root.mli

```diff
@@ -69,4 +78,36 @@ module Make
     -> src:t
     -> dest:t
     -> t Or_error.t
+
+  (** Retrieve the depth of the root ledger *)
+  val depth : t -> int
+
+  (** Retrieve the number of accounts in the root ledger *)
+  val num_accounts : t -> int
+
+  (** Calculate the address path from [addr] to the merkle root. *)
+  val merkle_path_at_addr_exn : t -> addr -> path
+
+  (** Get the hash at the given [addr] in the ledger. Throws an exception if the
+      [addr] doesn't point to a hash. *)
+  val get_inner_hash_at_addr_exn : t -> addr -> hash
+
+  (** Calculate all the addresses that lie in the ledger, starting at [addr] and
+      going downward in the merkle tree. Then, set the accounts at those
+      addresses to the values in the list. Throws an exception if the [addr] is
+      not in the ledger, or if there are not exactly enough accounts in the list
+      to set the values at all the addresses. *)
+
+  (* TODO: see if the hashes of the accounts can be passed in here too *)
```

Comment:
Why is there this TODO? Is this copy-pasted from somewhere else, or is there a specific use case for this functionality?

Reply:
It's a performance thing I noticed. In the implementation of `set_all_accounts_rooted_at_exn` on databases, this ends up calling `set_batch_accounts`, which has to hash all the accounts in the list to update the merkle tree structure in the database. If we happened to have the account hashes on hand where this is used, then we could avoid re-hashing the accounts.

It might be better to document this as an issue.

Reply:
Both the TODOs are original, but they don't really belong there